psycop/common/feature_generation/sequences/timeseries_windower_python/events/temporal_event.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import datetime as dt
+
+    from psycop.common.feature_generation.sequences.timeseries_windower_python.patient import (
+        Patient,
+    )
+
+
+@dataclass
+class TemporalEvent:
+    timestamp: dt.datetime
+    source: str  # E.g. "lab"/"diagnosis"
+    name: str | None  # E.g. "Hba1c"/"hypertension". Is optional, since e.g. diagnoses don't have a name, only a source and value.
+    value: float | str  # 1/0 for booleans, numeric value for numeric events
+    patient: Patient | None = None
+
+    def __eq__(self, __value: TemporalEvent) -> bool:  # type: ignore
+        # We implement a custom equality, since we have circular references in the dataclass,
```

Comment:
```suggestion
    def __eq__(self, __value: TemporalEvent) -> bool:  # type: ignore
        """Check if the event is the same. Does not check if the patient is the same"""
        # We implement a custom equality, since we have circular references in the dataclass,
```